src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -60,6 +61,7 @@
 
   private final String namespace;
   private Watch buildsWatch;
+  private ScheduledFuture relister;
 
   public BuildWatcher(String namespace) {
```

Comment:
One thing to consider is to extend BuildWatcher with either PeriodicWork or ASyncPeriodicWork. It removes having to inline the task and you might benefit from built-in housekeeping...just a thought.